src/main/resources/META-INF/rust-core.xml

```diff
@@ -1149,6 +1149,10 @@
         <filetype.prebuiltStubsProvider filetype="Rust"
                                         implementationClass="org.rust.lang.core.macros.MacroExpansionStubsProvider"/>
 
+        <applicationService serviceInterface="org.rust.lang.core.resolve2.ResolveCommonThreadPool"
+                            serviceImplementation="org.rust.lang.core.resolve2.ResolveCommonThreadPoolImpl"
+                            testServiceImplementation="org.rust.lang.core.resolve2.TestResolveCommonThreadPoolImpl"/>
```

Comment:
@vlad20012 maybe it's better to have a single implementation of thread pool creation in this case just to make sure that everything is fine in production as well
The corresponding error can be solved via:
- single service implementation and doing test code only things under `isUnitTestMode`
- still create test service implementation. But in addition, always set custom name for all created threads (I think it may be useful in general to understand what thread  does looking at thread dump, for example), and call `ThreadTracker.longRunningThreadCreated` in constructor of test service - it will be trivial since thread prefix will be custom